ui/effects/use-get-user-memberships.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+import { useState, useEffect } from 'react';
+import { getChannelFromClaim } from 'util/claim';
+
+export default function useGetUserMemberships(
+  shouldFetchUserMemberships: ?boolean,
+  arrayOfContentUris: Array<string>,
+  convertClaimUrlsToIds: any,
+  doFetchUserMemberships: (string) => void // fetch membership values and save in redux
+) {
+  const [fetchedUserClaims, setFetchedUserClaims] = useState([]);
+
+  useEffect(() => {
+    if (shouldFetchUserMemberships && arrayOfContentUris && arrayOfContentUris.length > 0) {
+      // check against the uris already saved in memory (already fetched)
+      const urisToFetch = arrayOfContentUris.filter(
+        (uri) => uri && !fetchedUserClaims.includes(uri) && Boolean(convertClaimUrlsToIds[uri])
+      );
+
+      if (urisToFetch.length > 0) {
+        // convert uris to claimIds
+        const claimIds = arrayOfContentUris.map((uri) => {
+          const claimUrlsToId = convertClaimUrlsToIds[uri];
+          if (claimUrlsToId) {
+            // TODO: flow error
+            return getChannelFromClaim(claimUrlsToId) && getChannelFromClaim(claimUrlsToId).claim_id;
+          }
+        });
```

Comment:
Several issues here.

(1)
Flow doesn't know you've checked for null, but it is also bad practice to call a function twice, no matter how trivial (it could become non-trivial when something is added to it later).  Stash it in a variable.
```
  const channelClaim = getChannelFromClaim(claimUrlsToId);
  return channelClaim && channelClaim.claim_id;
```

(2)
But if you need the channel claim ID, can use `getChannelIdFromClaim` directly.

(3)
But both functions are expecting a `Claim`, but you are passing an ID (`claimUrlsToId`)?